### ISSUE TITLE
DC instantiate rest storage: apiserver was unable to write a JSON response: http: request method or response status code does not allow body

### ISSUE THREAD

**Issue by ncdc** [openshift/origin#13986]
```
E0501 17:27:06.286767    7946 writers.go:113] apiserver was unable to write a JSON response: http: request method or response status code does not allow body
E0501 17:27:06.286788    7946 status.go:62] apiserver received an error that is not an metav1.Status: http: request method or response status code does not allow body
```

I've narrowed this down to this return statement: https://github.com/openshift/origin/blob/a71e6f14824d4b5bb3eb852104163b01cf32f1fb/pkg/deploy/registry/instantiate/rest.go#L80-L83

cc @kargakis @mfojtik @pweil- @deads2k @liggitt @sttts @soltysh @smarterclayton

**Comment by sttts** [comment]
https://httpstatuses.com/204

**Comment by smarterclayton** [comment]
Bad instantiate!  Bad!

Not sure what code it should be though - not 200 if we can find a
better one.  We don't have a "no-op" reason defined

**Comment by ncdc** [comment]
@sttts this was duplicated by #14213 and fixed by #14237, correct?

**Comment by sttts** [comment]
@deads2k I have my problems following the given statement https://github.com/openshift/origin/blob/a71e6f14824d4b5bb3eb852104163b01cf32f1fb/pkg/deploy/registry/instantiate/rest.go#L80-L83 to an actual 204 "on the wire". As far as I see, https://github.com/kubernetes/kubernetes/blob/1f45c4846bafaa8f2f17deb53f4284cc78e83210/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go#L446-L446 uses 201 explicitly and that doesn't get overwritten in the write.

About  #14237: it fixes the `ErrorNegotiated` writer, not a general `WriteObject(... &metav1.Status...)`. So I am note sure we caught this here already.
